### DISCUSSION TITLE
Release 9.3.0

### DISCUSSION THREAD

**Discussion by Cheesebaron** [MvvmCross/MvvmCross#4880] [General]
## What's Changed

It has been a while since last stable release, time for a new one. Some notable changes include:

- Removal of .NET 7 TFM
- Addition of .NET 9 TFM
- Lots of warnings cleaned up and enabled nullable attributes in more places
- Build fully moved to GitHub Actions
- Multi Window support for WinUI 3 with the help of @Joost-Jens-Luminis 

A big thank you to all the contributors, all help is appreciated.

### Breaking Changes 🛠
* Use structured logging by @Cheesebaron in https://github.com/MvvmCross/MvvmCross/pull/4827
* Big warnings cleanup by @Cheesebaron in https://github.com/MvvmCross/MvvmCross/pull/4839
* Remove .NET 7 TFMs by @Cheesebaron in https://github.com/MvvmCross/MvvmCross/pull/4844
### New Features 🎉
* Enable nullable on target bindings on Android and iOS by @Cheesebaron in https://github.com/MvvmCross/MvvmCross/pull/4826
* Enable nullable attributes in MvxRecyclerView lib by @Cheesebaron in https://github.com/MvvmCross/MvvmCross/pull/4835
* Added multi window support for WinUI 3 by @Joost-Jens-Luminis in https://github.com/MvvmCross/MvvmCross/pull/4819
### Fixed Bugs 🐛
* Run continuations from TaskCompletionSource on TaskPool by @Cheesebaron in https://github.com/MvvmCross/MvvmCross/pull/4864
* Fix issues with Window exports in AppDelegate by @Cheesebaron in https://github.com/MvvmCross/MvvmCross/pull/4873
### Dependencies 👒
* Bump cake.tool from 3.2.0 to 4.0.0 by @dependabot in https://github.com/MvvmCross/MvvmCross/pull/4775
* Bump the unittest-dependencies group with 2 updates by @dependabot in https://github.com/MvvmCross/MvvmCross/pull/4773
* Bump the analyzers group with 1 update by @dependabot in https://github.com/MvvmCross/MvvmCross/pull/4774
* Bump dotnet-sonarscanner from 5.14.0 to 5.15.0 by @dependabot in https://github.com/MvvmCross/MvvmCross/pull/4776
* Bump the analyzers group with 1 update by @dependabot in https://github.com/MvvmCross/MvvmCross/pull/4777
* Bump dotnet-sonarscanner from 5.15.0 to 6.0.0 by @dependabot in https://github.com/MvvmCross/MvvmCross/pull/4783
* Bump the analyzers group with 2 updates by @dependabot in https://github.com/MvvmCross/MvvmCross/pull/4782
* Bump the unittest-dependencies group with 2 updates by @dependabot in https://github.com/MvvmCross/MvvmCross/pull/4791
* Bump the unittest-dependencies group with 1 update by @dependabot in https://github.com/MvvmCross/MvvmCross/pull/4792
* Bump the analyzers group with 2 updates by @dependabot in https://github.com/MvvmCross/MvvmCross/pull/4795
* Bump the unittest-dependencies group with 1 update by @dependabot in https://github.com/MvvmCross/MvvmCross/pull/4794
* Bump Microsoft.Maui.Essentials from 8.0.3 to 8.0.6 by @dependabot in https://github.com/MvvmCross/MvvmCross/pull/4801
* Bump the analyzers group with 2 updates by @dependabot in https://github.com/MvvmCross/MvvmCross/pull/4800
* Bump the analyzers group with 1 update by @dependabot in https://github.com/MvvmCross/MvvmCross/pull/4802
* Bump dotnet-sonarscanner from 6.0.0 to 6.1.0 by @dependabot in https://github.com/MvvmCross/MvvmCross/pull/4806
* Bump the analyzers group with 2 updates by @dependabot in https://github.com/MvvmCross/MvvmCross/pull/4805
* Bump the unittest-dependencies group with 2 updates by @dependabot in https://github.com/MvvmCross/MvvmCross/pull/4811
* Bump Microsoft.Maui.Essentials from 8.0.6 to 8.0.7 by @dependabot in https://github.com/MvvmCross/MvvmCross/pull/4815
* Bump the analyzers group with 2 updates by @dependabot in https://github.com/MvvmCross/MvvmCross/pull/4809
* Bump the unittest-dependencies group with 2 updates by @dependabot in https://github.com/MvvmCross/MvvmCross/pull/4813
* Bump the unittest-dependencies group with 3 updates by @dependabot in https://github.com/MvvmCross/MvvmCross/pull/4817
* Bump the analyzers group with 1 update by @dependabot in https://github.com/MvvmCross/MvvmCross/pull/4818
* Bump dotnet-sonarscanner from 6.1.0 to 6.2.0 by @dependabot in https://github.com/MvvmCross/MvvmCross/pull/4821
* Bump the unittest-dependencies group with 2 updates by @dependabot in https://github.com/MvvmCross/MvvmCross/pull/4820
* Bump the analyzers group with 2 updates by @dependabot in https://github.com/MvvmCross/MvvmCross/pull/4822
* Bump Microsoft.WindowsAppSDK from 1.4.231008000 to 1.4.231115000 by @dependabot in https://github.com/MvvmCross/MvvmCross/pull/4772
* Bump the unittest-dependencies group with 1 update by @dependabot in https://github.com/MvvmCross/MvvmCross/pull/4824
* Bump the analyzers group with 1 update by @dependabot in https://github.com/MvvmCross/MvvmCross/pull/4825
* Bump the unittest-dependencies group with 1 update by @dependabot in https://github.com/MvvmCross/MvvmCross/pull/4828
### Other Changes 🌱
* Bugfix. ModalControllerDelegate.WillDismiss() closes a ViewController when the ViewController is not actually dismissed by a user by @evgenyvalavin in https://github.com/MvvmCross/MvvmCross/pull/4785
* Add build workflow by @Cheesebaron in https://github.com/MvvmCross/MvvmCross/pull/4829
* Add sonar workflow by @Cheesebaron in https://github.com/MvvmCross/MvvmCross/pull/4830
* Ensuring unoverridable code returns to the main thread... by @dsmitchell in https://github.com/MvvmCross/MvvmCross/pull/4834
* Fix sonar analysis by @Cheesebaron in https://github.com/MvvmCross/MvvmCross/pull/4841
* Add package readme by @Cheesebaron in https://github.com/MvvmCross/MvvmCross/pull/4842
* Documentation update for WinUI by @Joost-Jens-Luminis in https://github.com/MvvmCross/MvvmCross/pull/4847
* Fixed creating and loading an Activity VM and its lifecycle twice after restoring on Android by @entdark in https://github.com/MvvmCross/MvvmCross/pull/4849
* CloseWindow is now called when the Window is actually closed by @Joost-Jens-Luminis in https://github.com/MvvmCross/MvvmCross/pull/4852
* Add Property information to SetValue exception by @JimWilcox3 in https://github.com/MvvmCross/MvvmCross/pull/4861
* Add .NET 9.0 TFM by @Cheesebaron in https://github.com/MvvmCross/MvvmCross/pull/4869
* Fixed verbiage in the readme.txt that gets presented when MvvmCross is installed. by @JimWilcox3 in https://github.com/MvvmCross/MvvmCross/pull/4875
* Update directory props and targets by @Cheesebaron in https://github.com/MvvmCross/MvvmCross/pull/4877
* Add workflow to label PRs automatically by @Cheesebaron in https://github.com/MvvmCross/MvvmCross/pull/4879
* Move .NET version to env var + pin when restoring workloads by @Cheesebaron in https://github.com/MvvmCross/MvvmCross/pull/4878

## New Contributors
* @dsmitchell made their first contribution in https://github.com/MvvmCross/MvvmCross/pull/4834
* @Joost-Jens-Luminis made their first contribution in https://github.com/MvvmCross/MvvmCross/pull/4819

**Full Changelog**: https://github.com/MvvmCross/MvvmCross/compare/9.2.0...9.3.0

<hr /><em>This discussion was created from the release <a href='https://github.com/MvvmCross/MvvmCross/releases/tag/9.3.0'>Release 9.3.0</a>.</em>

**Comment by Leo506** [comment]
Why was `MvxModalPresentationControllerDelegate` made sealed?

**Reply by Cheesebaron** [reply]
You can still override: `IUIAdaptivePresentationControllerDelegate CreateModalPresentationControllerDelegate(UIViewController viewController, MvxModalPresentationAttribute attribute)` in `MvxIosViewPresenter` and provide your own implementation.
